### ISSUE TITLE
`multi` with a block does not return errors in array

### ISSUE THREAD

**Issue by kazutomi** [redis/redis-rb#602]
Without block, errors are found in the return value (array) from `exec`:

```
> redis.set('akey', 'a string')
=> "OK"
> redis.multi
=> "OK"
> redis.sadd('akey', 'another')
=> false
> redis.incr('akey')
=> "QUEUED"
> redis.exec
=> [#<Redis::CommandError: WRONGTYPE Operation against a key holding the wrong kind of value>, #<Redis::CommandError: ERR value is not an integer or out of range>]
```

When used with a block, the error does not appear in the return value...

```
> redis.multi do |m|
>   m.sadd('akey', 'another')
> end  
=> [false]
```

...or it is just raised (i.e., not returned):

```
> redis.multi do |m|
>   m.incr('akey')
> end  
Redis::CommandError: ERR value is not an integer or out of range
```

The former behavior (without block) looks more consistent, except that `sadd` returns false ("QUEUED" expected); I found some other commands like `setnx` and `expire` also return false when queued.

**Comment by badboy** [comment]
`SADD` returning false in a transaction is indeed unexpected. It's because we don't keep track that we're in a transaction and [boolify](https://github.com/redis/redis-rb/blob/1ae5ce574125da8df88be233f53f32b4bde755d3/lib/redis.rb#L1293) any response unconditionally. Same for some others.

**Comment by byroot** [comment]
Fixed in #754
